lightning-macros/src/lib.rs

```diff
@@ -74,3 +77,219 @@ pub fn maybe_await(expr: TokenStream) -> TokenStream {
+			// We explicitly check the :s, =, and >, as well as an optional trailing ,
+			let enum_ident = arm_stream.next().unwrap();
+			let co1 = arm_stream.next().unwrap();
+			expect_punct(&co1, ':');
+			let co2 = arm_stream.next().unwrap();
+			expect_punct(&co2, ':');
+			let variant_ident = arm_stream.next().unwrap();
+			let fields = arm_stream.next().unwrap();
+			let eq = arm_stream.next().unwrap();
+			expect_punct(&eq, '=');
+			let gt = arm_stream.next().unwrap();
+			expect_punct(&gt, '>');
+			let init = arm_stream.next().unwrap();
+
+			let next_tok = arm_stream.peek();
+			if let Some(TokenTree::Punct(_)) = next_tok {
+				expect_punct(next_tok.unwrap(), ',');
+				arm_stream.next();
+			}
+
+			let computed_fields = if let TokenTree::Group(group) = fields {
+				process_fields(group)
+			} else {
+				panic!("Expected a group for the fields in a match arm");
+			};
+
+			let arm_pfx = [enum_ident, co1, co2, variant_ident];
+			new_arms.extend(arm_pfx.into_iter().map(token_to_stream));
+			new_arms.extend(computed_fields);
+			let arm_sfx = [eq, gt, init];
+			new_arms.extend(arm_sfx.into_iter().map(token_to_stream));
+		}
+
+		let new_arm_group = Group::new(Delimiter::Brace, new_arms);
+		res.extend(token_to_stream(TokenTree::Group(new_arm_group)));
+	} else {
+		panic!("Expected `match self {{..}}` and nothing else");
+	}
+
+	assert!(stream.next().is_none(), "Expected `match self {{..}}` and nothing else");
+
+	res
+}
+
+/// Scans an enum definition for fields initialized with `legacy` types and drops them.
+///
+/// This is used internally in LDK's TLV serialization logic and is not expected to be used by
+/// other crates.
+///
+/// Is expected to wrap a struct definition like
+/// ```ignore
+/// drop_legacy_field_definition!(Self {
+///		field1: _init_tlv_based_struct_field!(field1, option),
+///     field2: _init_tlv_based_struct_field!(field2, (legacy, u64, {}, {})),
+/// })
+/// ```
+/// and will drop fields defined like `field2` with a type starting with `legacy`.
+#[proc_macro]
+pub fn drop_legacy_field_definition(expr: TokenStream) -> TokenStream {
+	let mut st = if let Ok(parsed) = parse::<syn::Expr>(expr) {
+		if let syn::Expr::Struct(st) = parsed {
+			st
+		} else {
+			return (quote! {
+				compile_error!("drop_legacy_field_definition!() can only be used on struct expressions")
+			})
+			.into();
+		}
+	} else {
+		return (quote! {
+			compile_error!("drop_legacy_field_definition!() can only be used on expressions")
+		})
+		.into();
+	};
+	assert!(st.attrs.is_empty());
+	assert!(st.qself.is_none());
+	assert!(st.dot2_token.is_none());
+	assert!(st.rest.is_none());
+	let mut new_fields = syn::punctuated::Punctuated::new();
+	core::mem::swap(&mut new_fields, &mut st.fields);
+	for field in new_fields {
+		if let syn::Expr::Macro(syn::ExprMacro { mac, .. }) = &field.expr {
+			let macro_name = mac.path.segments.last().unwrap().ident.to_string();
+			let is_init = macro_name == "_ignore_arg";
```

Comment:
I somehow would think that the `ignore_macro` would be expanded before the `proc_macro` would be able to view the macro name, if that makes sense. Any rules of thumb for how the macros get expanded? Tried to read this https://rustc-dev-guide.rust-lang.org/macro-expansion.html but it's a lot to take in